icu4c/source/common/unicode/unistr.h

```diff
@@ -105,13 +105,11 @@ class UnicodeStringAppendable;  // unicode/appendable.h
  * this macro was provided for portability and efficiency when
  * initializing UnicodeStrings from literals.
  *
- * Since C++17 and ICU 76, you can use std::u16string_view literals with compile-time
+ * Since C++17 and ICU 76, you can use UTF-16 string literals with compile-time
  * length determination:
  * \code
- * #include &lt;string_view&gt;
- * using namespace std::string_view_literals;
- * UnicodeString str(u"literal"sv);
- * if (str == u"other literal"sv) { ... }
+ * UnicodeString str(u"literal");
+ * if (str == u"other literal") { ... }
```

Comment:
Please revert the operator== example. Users who make the constructor explicit don't get the vanilla literal auto-converted to a UnicodeString, and users who do get the auto conversion get a copy of the string literal contents. With your change here, they would avoid strlen() but not the contents copy.

Using the new operator==(const S &) would still be more efficient with an input u16string_view.

Reply:
I guess that either I don't understand what you mean, or the compiler doesn't do what you think it does. To verify what the compiler really does, I wrote this piece of example code:

```c++
bool is_foo(const icu::UnicodeString& input) {
  return input == u"foo";
}
```

And this piece of example code, using an inlined `std::u16string_view` instead of the UTF-16 string literal:

```c++
bool is_foo(const icu::UnicodeString& input) {
  return input == u"foo"sv;
}
```

Testing with the Clang 18 compiler, these both result in the same, both get the known length hardcoded and this length together with a pointer to the string literal then gets passed directly to the underlying `doEquals()` function.

This is the behaviour that I had expected and the reason for why I proposed updating the documentation to recommend the simpler code that uses the string literal directly. Did you expect something else, or am I just not understanding why you prefer using `u""sv` instead?

Reply:
Maybe this could become easier to understand by flipping the question around:

Why would `operator==(const UnicodeString&)` be invoked unless you already have a `UnicodeString` object?

The compiler knows how to directly make a `std::u16string_view` from a `u""` string literal.

Reply:
Ok. I understand that my assumption was wrong. A raw pointer or array is ConvertibleToU16StringView. I need to get my head around all of the effects from that.

Reply:
FYI The simplification of API docs here is timely after my recent PR that introduced those, but not related to the constructor changes in this PR. I think that's part of what tripped me up.

Reply:
> The simplification of API docs here is timely after my recent PR that introduced those, but not related to the constructor changes in this PR.

Oh, is it not? After I learned (PR #3102) that the constructor required passing it a string view in order to not recalculate the length, I just assumed that the reason for the documentation to recommend always passing a string view (even for functions where it wasn't actually needed) was so that people wouldn't have to keep in mind whether they were invoking the constructor or some other function, so after fixing the constructor it seemed obvious to me to also update this part of the documentation at the same time as I then no longer could see any advantage in continuing to recommend the use of string views instead of string literals.

Reply:
The simplification of the constructor call site is related to the constructor change in this PR.
The simplification of the operator== call site didn't require the constructor change, because it doesn't construct a UnicodeString.
Not having to think about when to use which is even better.